### PR TITLE
Adds skeleton project with dummy streaming job

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds a skeleton project which separates the how properties are
supplied from code that uses it. For example, the core library has
no spring dependencies, or other property management utilities.

This borrows the maven configuration from zipkin-dependencies so that
it can be used in spark clusters. Note this isn't fully tested, rather
a scratch pad for others to contribute to.

### CODE REVIEW

File: sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import scala.Tuple2;
+import zipkin.Codec;
+import zipkin.Span;
+import zipkin.internal.Util;
+
+import static zipkin.internal.Util.checkArgument;
+import static zipkin.internal.Util.checkNotNull;
+
+public final class SparkStreamingJob implements Closeable {
```
Comment (adriancole) [inline sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java]: for those scrolling through maven, this is the interesting part

File: sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    Map<String, String> sparkProperties = new LinkedHashMap<>();
+    long batchDuration = 10 * 1000;
+
+    Builder() {
+      sparkProperties.put("spark.ui.enabled", "false");
+      // avoids strange class not found bug on Logger.setLevel
+      sparkProperties.put("spark.akka.logLifecycleEvents", "true");
+    }
+
+    /** The time interval at which streaming data will be divided into batches. Defaults to 10s. */
```
Comment (devinsba) [inline sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java]: This comment is not for this method

**Review by devinsba — COMMENTED** [review]
I like the approach, would be interesting to see a sample implementation of the span stream

File: sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java (hunk 2 above)
Comment (adriancole) [inline sparkstreaming/src/main/java/zipkin/sparkstreaming/SparkStreamingJob.java]: fixed. thanks

**Pull request merged**